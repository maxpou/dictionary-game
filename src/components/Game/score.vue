<template>
  <div v-if="totalScorePercentage" class="ui indicating progress">
    <div class="bar" v-bind:style="{ width: totalScorePercentage + '%' }">
      <div class="progress">{{ score.success }} / {{ this.score.success + this.score.errors }} ({{ totalScorePercentage }}%)</div>
    </div>
  </div>
</template>

<script>
export default {
  props: ['score'],
  computed: {
    totalScorePercentage () {
      return Math.floor(100 * this.score.success / (this.score.success + this.score.errors))
    }
  }
}
</script>

<style scoped>
.ui.progress .bar {
  background-color: #daa520;
}
</style>
