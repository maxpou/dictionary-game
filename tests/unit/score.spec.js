import { mount } from '@vue/test-utils'
import score from '@/components/Game/score.vue'

describe('score.vue', () => {
  it('shouldn\'t render when empty score', () => {
    const wrapper = mount(score, {
      propsData: {
        score: {
          success: 0,
          errors: 0
        }
      }
    })
    const template = wrapper.html()
    expect(template).toBeUndefined()
  })

  it('should render component', () => {
    const wrapper = mount(score, {
      propsData: {
        score: {
          success: 23,
          errors: 5
        }
      }
    })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
