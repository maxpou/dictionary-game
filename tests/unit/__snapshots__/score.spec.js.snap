// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`score.vue should render component 1`] = `
<div class="ui indicating progress">
  <div class="bar" style="width: 82%;">
    <div class="progress">23 / 28 (82%)</div>
  </div>
</div>
`;
